src/Fixer/Semicolon/MultilineWhitespaceBeforeSemicolonsFixer.php

```diff
@@ -0,0 +1,290 @@
+
+            // appended new line to the last method call
+            $newline = new Token([T_WHITESPACE, $lineEnding.$indent]);
+
+            // insert the new line with indented semicolon
+            $tokens->insertAt($index, [$newline, new Token(';')]);
+        }
+    }
+
+    /**
+     * Find the index for the new line. Return the given index when there's no new line.
+     *
+     * @param int $index
+     *
+     * @return int
+     */
+    private function getNewLineIndex($index, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        for ($index, $count = count($tokens); $index < $count; ++$index) {
+            if (false !== strstr($tokens[$index]->getContent(), $lineEnding)) {
+                return $index;
+            }
+        }
+
+        return $index;
+    }
+
+    /**
+     * Checks if the semicolon closes a chained call and returns the whitespace of the first call at $index.
+     * i.e. it will return the whitespace marked with '____' in the example underneath.
+     *
+     * ..
+     * ____$this->methodCall()
+     *          ->anotherCall();
+     * ..
+     *
+     * @param int $index
+     *
+     * @return null|string
+     */
+    private function findWhitespaceBeforeFirstCall($index, Tokens $tokens)
+    {
+        // semicolon followed by a closing bracket?
+        if (!$tokens[$index]->equals(')')) {
+            return null;
+        }
+
+        // find opening bracket
+        $openingBrackets = 1;
+        for (--$index; $index > 0; --$index) {
+            if ($tokens[$index]->equals(')')) {
+                ++$openingBrackets;
+
+                continue;
+            }
+
+            if ($tokens[$index]->equals('(')) {
+                if (1 === $openingBrackets) {
+                    break;
+                }
+                --$openingBrackets;
+            }
+        }
+
+        // method name
+        if (!$tokens[--$index]->isGivenKind(T_STRING)) {
+            return null;
+        }
+
+        // ->
+        if (!$tokens[--$index]->isGivenKind(T_OBJECT_OPERATOR)) {
+            return null;
+        }
+
+        // white space
+        if (!$tokens[--$index]->isGivenKind(T_WHITESPACE)) {
+            return null;
+        }
+
+        $closingBrackets = 0;
+        for ($index; $index >= 0; --$index) {
+            if ($tokens[$index]->equals(')')) {
+                ++$closingBrackets;
+            }
+
+            if ($tokens[$index]->equals('(')) {
+                --$closingBrackets;
+            }
+
+            // must be the variable of the first call in the chain
+            if ($tokens[$index]->isGivenKind(T_VARIABLE) && 0 === $closingBrackets) {
+                if ($tokens[--$index]->isGivenKind(T_WHITESPACE)
+                    || $tokens[$index]->isGivenKind(T_OPEN_TAG)) {
+                    return $this->getIndentAt($tokens, $index);
+                }
+            }
+        }
+
+        return null;
+    }
```

Comment:
It would make sense to decouple this and following method to standalone class, with decopuled tests, as it detects lines indentation. Name it e.g. `WhitespaceIndentDetector`.
Few more fixer would use it and other people in their fixers too, instead of re-writing code again.

Not sure where is the right place to put it though @keradus @SpacePossum

Reply:
Nice idea! Let me know @keradus @SpacePossum if it's in the scope of this PR or should it be changed separately?

Reply:
separated PR for sure

Reply:
Cool! I've created an issue for that https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3416